Sources/TSCBasic/Process.swift

```diff
@@ -53,7 +144,7 @@ public struct ProcessResult: CustomStringConvertible, Sendable {
     public let arguments: [String]
 
     /// The environment with which the process was launched.
-    public let environment: [String: String]
+    public let environment: ProcessEnvironmentBlock
```

Comment:
This is a breaking API change, while new API surface has to be introduced in additive manner, with older API deprecated first. Otherwise this will break clients of TSC that we don't have direct control over.

Reply:
Done